compiler/jsgen.nim

```diff
@@ -179,9 +179,36 @@ proc mapType(p: PProc; typ: PType): TJSTypeKind =
   else: result = mapType(typ)
 
 proc mangleName(s: PSym; target: TTarget): Rope =
+  proc validJsName(name: string): bool =
+    result = true
+    const reservedWords = ["abstract", "await", "boolean", "break", "byte",
```

Comment:
sorry but I said to use a string case statement here. That will be compiled into a compiletime hash table. Easier, faster.